src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -1640,5 +1640,38 @@ public void Test_Issue2268()
             evaluated.Should().HaveValueAtPath("$.resources[0].name", "myServer/myDb/myTde");
             evaluated.Should().HaveValueAtPath("$.outputs['tdeId'].value", "/subscriptions/f91a30fd-f403-4999-ae9f-ec37a6d81e13/resourceGroups/testResourceGroup/providers/Microsoft.Sql/servers/myServer/databases/myDb/transparentDataEncryption/myTde");
         }
+
+        [TestMethod]
+        // https://github.com/azure/bicep/issues/1809
+        public void Test_Issue1809()
+        {
+            var result = CompilationHelper.Compile(
+                ("main.bicep", @"
+module tags './tags.bicep' = {
+  name: 'tags'
+}
+
+resource vwan 'Microsoft.Network/virtualWans@2020-05-01' = {
+  location: 'westus'
+  name: 'vwan'
+  properties: {
+    disableVpnEncryption: false
+    allowBranchToBranchTraffic: true
+    allowVnetToVnetTraffic: true
+    type: 'foo'
+  }
+  tags: tags.outputs.tagsoutput
+}
+"),
+                ("tags.bicep", @"
+output tagsoutput object = {
+  tag1: 'tag1Value'
+}
+"));
+
+            result.Should().HaveDiagnostics(new[] {
+                ("BCP120", DiagnosticLevel.Error, "The property \"tags\" must be evaluable at the start of the deployment, and cannot depend on any values that have not yet been calculated. Accessible properties of tags are \"name\", \"scope\"."),
```

Comment:
@anthony-c-martin  @majastrz question - I've noticed that we currently mark `scope` as write-only when create `ModuleType`, so it cannot be referenced even it is a deploy-time constant and it shows up in the error message. Is there any specific reason to make it write-only?

Reply:
We probably didn't want to codegen that, but it's weird that the message is showing it. Are missing a filter when constructing the list of accessible properties?

Reply:
Got it. Yeah we should filter out write-only properties. I'll push a commit to fix it.

Reply:
Done.

Reply:
Nice!